### PR TITLE
send absence data to back-end; interactive buttons

### PR DESCRIPTION
**Pull request by vladisa88**

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+name = "pypi"
+url = "https://pypi.org/simple"
+verify_ssl = true
+
+[dev-packages]
+
+[packages]
+
+[requires]
+python_version = "3.9"
```
Comment (TmLev) [inline Pipfile:11]: Redundant file.

File: tgbot/src/api/absence.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        data: dict[str, tp.Any] = await response.json()
+    return data
+
+
+async def post_all_absence(students: list[Student]) -> bool:
+    results = await asyncio.gather(
+        *[post_one_absence(student) for student in students
+          if not bool(int(student.state))],
+        return_exceptions=True)
+    return absence_statistic(students)
```
Comment (TmLev) [inline tgbot/src/api/absence.py:51]: Synchronous waiting, instantaneous gather.

File: tgbot/src/api/student.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+from aiohttp import ClientResponse
+
+from api.client import client
+
+
+@dataclass
+class Student:
+    id: int
+    full_name: str
+    state: int = 1
```
Comment (TmLev) [inline tgbot/src/api/student.py:15]: Better use enum.

**Review by TmLev — CHANGES_REQUESTED** [review]

**Comment by gakhromov** [comment]
delet this

**Pull request closed without merging**